includes/collection/class-users.php

```diff
@@ -96,9 +98,14 @@ public static function get_by_username( $username ) {
 				'meta_query'  => array(
 					'relation' => 'OR',
 					array(
-						'key'     => 'activitypub_user_identifier',
+						'key'     => 'activitypub_identifier',
+						'value'   => $username,
+						'compare' => '=',
+					),
+					array(
```

Comment:
It's probably not worth addressing but it could happen that you assign the identifier to a non-existent user at the time, and then register that username afterwards. In this scenario I believe the result here is undefined since you just pick the first result in the code below.